### PR TITLE
Marked commonly used interfaces as implicitely used

### PR DESCRIPTION
**Pull request by danielmarbach**
@Particular/engineers Any objections?

### CODE REVIEW

File: src/NServiceBus.Core/Hosting/Configuration/IConfigureThisEndpoint.cs [hunk 1]
```diff
@@ -1,15 +1,17 @@
 ﻿namespace NServiceBus
 {
+    using JetBrains.Annotations;
+
     /// <summary>
     /// Indicate that the implementing class will specify configuration.
     /// </summary>
+    [UsedImplicitly(ImplicitUseTargetFlags.WithMembers)]
```
Comment (SimonCropp) [inline src/NServiceBus.Core/Hosting/Configuration/IConfigureThisEndpoint.cs]: this one needs to be moved to the host

**Comment by SimonCropp** [comment]
I am confused. as per the usage of the attribute

> Indicates that the marked symbol is used implicitly (e.g. via reflection, in external library), so this symbol will not be marked as unused (as well as by other usage inspections)

So any of these interfaces that are not explicitly used should highlight the fact that we have missing tests.

**Comment by johnsimons** [comment]
What are these annotations good for ?
Do we really need them ?

**Comment by danielmarbach** [comment]
Some of these interfaces are scanned and registered in the container.
Marking them means resharper won't mark it as obsolete

On Monday, March 9, 2015, John Simons <notifications@github.com> wrote:

> What are these annotations good for ?
> Do we really need them ?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Particular/NServiceBus/pull/2679#issuecomment-77950239>
> .
>


-- 

Daniel Marbach

Solution Architect, Particular Software

Email:  daniel.marbach@particular.net

Web:    www.particular.net
<http://t.signauxneuf.com/e1t/c/5/f18dQhb0S7lC8dDMPbW2n0x6l2B9nMJN7t5XX4RzDjlVS9QB88pTc_dW3Ljg7x56dWyBdgBXs402?t=http%3A%2F%2Fwww.particular.net%2F&si=4896787407044608&pi=33cc6202-1f31-4040-8d6c-e70641f44bba>

**Comment by johnsimons** [comment]
> Marking them means resharper won't mark it as obsolete

What? Is there any doco about it ?

**Comment by SimonCropp** [comment]
@johnsimons https://www.jetbrains.com/resharper/webhelp70/Reference__Code_Annotation_Attributes.html#UsedImplicitlyAttribute

**Comment by johnsimons** [comment]
Sorry guys, still very confused why we need this now ?

**Comment by danielmarbach** [comment]
It is not that we desperately need it. It would be just a good hint to
people using Resharper because then it would never mark the implementers of
these interfaces as not being actively used

**Comment by danielmarbach** [comment]
2015-03-10 14_40_03-nservicebus - microsoft visual studio

@johnsimons that wouldn't happen when we use those attributes

File: src/NServiceBus.Core/IAuthorizeSubscriptions.cs [hunk 2]
```diff
@@ -10,6 +10,7 @@ namespace NServiceBus
     /// 
     /// Infrastructure automatically registers one implementing type in the container as a singleton.
     /// </summary>
+    [UsedImplicitly(ImplicitUseTargetFlags.WithMembers)]
```
Comment (SimonCropp) [inline src/NServiceBus.Core/IAuthorizeSubscriptions.cs]: i dont understand this one. IAuthorizeSubscriptions and its members are used

**Comment by johnsimons** [comment]
As I see it, if there are public interfaces/classes that are not being used, IMO that should be addressed by creating a test that exercises that code, not by annotating it to make it go away.

@danielmarbach any objections against closing this one ?

**Comment by danielmarbach** [comment]
Talked this through with @johnsimons today and we agreed to pull it in because it provides benefits for endusers.

**Pull request closed without merging**